### PR TITLE
BL-705/connect mentor drop downs

### PR DESCRIPTION
**Pull request by khaleel737**
## Description

Added Mentor selection on the Matching Page of underdog, each user gets the mentors assigned based on the assignment and the mentor ID in the assignment object.

Fixes # (issue)

cleaned allot of code and refactored some, the code base had allot of issues in it, props not being passed correctly and using dummy data on the wrong page.

## Loom Video

https://www.loom.com/share/e9096227a5314e0a93ad9988e7a4274d

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/pages/MentorMenteeMatching/MentorMenteeInfo.js [hunk 1]
```diff
@@ -19,11 +20,17 @@ const MentorMenteeInfo = props => {
...
-                <p>{`Email: ${data.email}`}</p>
+                <p>{`Email: ${data.contact}`}</p>
                 <p>{`Location: ${data.location}`}</p>
                 <p>{`Company: ${data.company}`}</p>
               </Card>
 
+              {/**
+               * Owner: Khaleel Musleh
+               * Props Being Passed:
+               * Props: data={data} From MentorMenteeMatching to MentorsCard
```
Comment (Johnjx) [inline src/components/pages/MentorMenteeMatching/MentorMenteeInfo.js:31]: Great format, I will keep this in mind!

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 2]
```diff
@@ -1,21 +1,84 @@
...
+             * Owner: Khaleel Musleh
+             * Selection below gets a list from the GET api call above, then it iterates on each one then pussing an Option element with the first name,
+             * last name as values as well as seperate keys obrained from the index of the Array.
+             * All <Option> elements are pushed to the selections variable
+             */}
+
+            {mentors.forEach((selection, index) => {
+              selections.push(
+                <Option
+                  key={index}
```
Comment (Johnjx) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:71]: Nice form filling! I want to leave a note about keys here. I'm not sure if it was possible in your work but there are other ways to generate unique keys that can be more preferable, such as unique ids on items from the Database. See this post for [more](https://stackoverflow.com/questions/39549424/how-to-create-unique-keys-for-react-elements)!

**Review by Johnjx — APPROVED** [review]
Awesome work Khaleel. Getting your bearings around a new project is difficult & comments such as these are super helpful to your fellow developers.  It's definitely not a good feeling when your work is underappreciated so I want to say I really appreciate all your effort in digging in and improving the codebase!

**Review by nathancai97 — DISMISSED** [review]
Nice job implementing the drop downs. And also cleaning up old code!

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 3]
```diff
@@ -1,21 +1,84 @@
 import { Select } from 'antd';
+import { useEffect, useState } from 'react';
+import axiosWithAuth from '../../../utils/axiosWithAuth';
+
+export const Selected = (edit, setMentor, mentor, data, name, nameVal) => {
+  /**
+   * Owner: Khaleel Musleh
+   * Props Being Passed:
+   * Props: mentor parameter passed all the way from MentorMenteeMatching -> MentorMenteeInfo -> MentorsCard -> MentorsOptions
+   */
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:10]: I agree with @Johnjx that this is a good format to helpfully add contextual information and explanations.

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 4]
```diff
@@ -1,21 +1,84 @@
...
+   * Owner: Khaleel Musleh
+   * Selection variable has all the HTML elements to pass to the list of selections for each user, if a user has no mentors then it will return no data.
+   *
+   */
+  let selections = [];
+
+  const { Option } = Select;
+  /**
+   * Owner: Khaleel Musleh
+   * GET mentors based on the mentor_id found in the mentor props, it passes all the ID's into the GET function below and gets the mentors of each user seperately.
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:24]: There's just a small typo with what should be "separated" here.

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 5]
```diff
@@ -1,21 +1,84 @@
...
+   * setMentor gets the data of the mentor information then its passed below to selection
+   */
+  useEffect(() => {
+    axiosWithAuth()
+      .get(`/assignments/mentee/${mentor.key}`)
+      .then(response => {
+        setMentors(response.data);
+      });
+  }, []);
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:33]: Good job not forgetting that dependency array!

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 6]
```diff
@@ -1,21 +1,84 @@
...
+              option.props.children
+                .toLowerCase()
+                .indexOf(input.toLowerCase()) >= 0
+            }
           >
-            {options(data)}
+            {/**
+             * Owner: Khaleel Musleh
+             * Selection below gets a list from the GET api call above, then it iterates on each one then pussing an Option element with the first name,
+             * last name as values as well as seperate keys obrained from the index of the Array.
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:64]: There are three minor typos across two lines here with what should be "pushing", "separate", and "obtained".

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 7]
```diff
@@ -1,21 +1,84 @@
...
+             * All <Option> elements are pushed to the selections variable
+             */}
+
+            {mentors.forEach((selection, index) => {
+              selections.push(
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:69]: Good adaptable / flexible programmatic solution here to replace a more brittle hard-coded solution!

**Review by albert-szabo — APPROVED** [review]
Overall, these look to be good, solid changes. The consistent explanatory comments are a considerate addition.

**Review by MelodyLayne — APPROVED** [review]
You saw my comments and suggestions on the previous PR so I won't duplicate them here.  The only note I'll make is that you can still push code to an open PR and then re-request reviews. That'll save you from having to do a whole new one all over again. Still, I appreciate your diligent efforts to make sure the code is at it's best when merged into main!

**Review by brimjustin — DISMISSED** [review]
With all the changes, the update looks good and will be beneficial for the stakeholder.

**Review by miguelaledesma — DISMISSED** [review]
Great job implemented the drop down. Clean code goes a long way so thank you for adding notes and not leaving any commented code that does not belong. Also, thanks for the detailed comments. It will really help the next devs working on this project.

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 8]
```diff
@@ -1,21 +1,84 @@
...
+           * Form has filter option and removed the onChange to a seperated Function.
+           */}
           <Select
-            name={nameVal}
-            onChange={value => setMentors({ ...mentors, [nameVal]: value })}
             showSearch
-            style={{ width: '100%' }}
-            optionFilterProp="items"
-            value={name || 'Assign Mentor'}
+            style={{ width: 200 }}
```
Comment (zdravoj) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:51]: Please confirm this styling change - the rest of the code uses percentages so make sure this hard-coded value doesn't cause any problems!

**Review by zdravoj — DISMISSED** [review]
Looks good, great work. Settling into this project is tough but you've done well. Confirm the styling change (see the comment), other than that, nice job!

**Review by Nicholas-R-Payne — APPROVED** [review]
Well done with the dropdown implementation!

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 9]
```diff
@@ -1,21 +1,84 @@
...
+                  value={selection.first_name + ' ' + selection.last_name}
+                >
+                  {selection.first_name + ' ' + selection.last_name}
+                </Option>
+              );
+            })}
+
+            <>{selections}</>
           </Select>
+          ,
```
Comment (ElijahHopkin) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:81]: what does this comma do for the code? If nothing, then delete it.

File: src/components/pages/MentorMenteeMatching/MentorOptions.js [hunk 10]
```diff
@@ -26,19 +89,6 @@ export const select = (edit, setMentors, mentors, data, name, nameVal) => {
...
-export const options = data => {
-  const { Option } = Select;
-
-  return (
-    <>
-      <Option value="Assign Mentor">Assign Mentor</Option>
-      <Option value={data.first}>{data.first}</Option>
-      <Option value={data.second}>{data.second}</Option>
-      <Option value={data.third}>{data.third}</Option>
-    </>
```
Comment (ElijahHopkin) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:38]: I like the flexibility of your solution a lot more. It is more code, but much more durable.

**Review by ElijahHopkin — APPROVED** [review]

File: src/components/pages/MentorMenteeMatching/MentorMenteeInfo.js (hunk 1 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorMenteeInfo.js:31]: Thank you

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 3 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js:10]: Thank you

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 5 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js:33]: thank you

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 6 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js:64]: Will resolve them today

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 7 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js:69]: Thank you

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 9 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js:81]: will do that

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 10 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js:38]: thank you

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 8 above)
Comment (paulstgermain) [inline src/components/pages/MentorMenteeMatching/MentorOptions.js:51]: Good changes here so far, but let's fix up this styling concern before we merge this up.

**Review by paulstgermain — CHANGES_REQUESTED** [review]

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 8 above)
Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js:51]: will fix it to 100% and push it now

Comment (khaleel737) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js]: I made a commit with the changes requested, fixed style width={"100%"} and spelling mistakes @paulstgermain

**Review by Nicholas-R-Payne — APPROVED** [review]
Resubmitting my review, still good job with the dropdown implementation!

**Review by MelodyLayne — APPROVED** [review]

File: src/components/pages/MentorMenteeMatching/MentorOptions.js (hunk 9 above)
Comment (ElijahHopkin) [reply src/components/pages/MentorMenteeMatching/MentorOptions.js]: great, Thanks!

**Review by ElijahHopkin — APPROVED** [review]

**Review by paulstgermain — APPROVED** [review]

**Pull request merged**